qiskit_nature/logging.py

```diff
@@ -0,0 +1,140 @@
+            add_default_handler: add or not the default stream handler
+        Returns:
+            Dictionary mapping names to effective level
+        """
+        names: List[str] = []
+        for name, level in name_levels.items():
+            names.append(name)
+            logger = self._logging.getLogger(name)
+            logger.setLevel(level)
+            logger.propagate = False
+            handlers = logger.handlers
+            if add_default_handler and not any(
+                True for h in handlers if isinstance(h, DefaultStreamHandler)
+            ):
+                self.add_handler(name, DefaultStreamHandler())
+
+        return self.get_levels_for_names(names)
+
+    def add_handler(
+        self, name: str, handler: logging.Handler, formatter: Optional[logging.Formatter] = None
+    ) -> None:
+        """Add handler and optional formatter to a module name.
+
+        Args:
+            name: module name
+            handler: Logging Handler
+            formatter: Logging Formatter
+        """
+        logger = self._logging.getLogger(name)
+        if formatter is not None:
+            handler.setFormatter(formatter)
+        logger.addHandler(handler)
+
+    def remove_handler(self, name: str, handler: logging.Handler) -> None:
+        """Remove handler from module.
+
+        Args:
+            name: module name
+            handler: Logging Handler
+        """
+        logger = self._logging.getLogger(name)
+        handler.close()
+        logger.removeHandler(handler)
+
+    def remove_all_handlers(self, names: List[str]) -> None:
+        """Remove all handlers from modules.
+
+        Args:
+            names: list of module names (qiskit_nature.drivers, qiskit.algorithms etc)
+        """
+        for name in names:
+            logger = self._logging.getLogger(name)
+            handlers = logger.handlers.copy()
+            for handler in handlers:
+                self.remove_handler(name, handler=handler)
+
+    def remove_default_handler(self, names: List[str]) -> None:
+        """Remove default handler from modules.
+
+        Args:
+            names: list of module names (qiskit_nature.drivers, qiskit.algorithms etc)
+        """
+        for name in names:
+            logger = self._logging.getLogger(name)
+            handlers = logger.handlers.copy()
+            for handler in handlers:
+                if isinstance(handler, DefaultStreamHandler):
+                    self.remove_handler(name, handler=handler)
+
+    def log_to_file(self, names: List[str], path: str) -> None:
+        """Logs modules to file.
+
+        Args:
+            names: list of module names (qiskit_nature.drivers, qiskit.algorithms etc)
+            path: file path
+        """
+        filepath = os.path.expanduser(path)
+        handler = logging.FileHandler(filepath, mode="w")
+        formatter = logging.Formatter(fmt="%(asctime)s:%(name)s:%(levelname)s: %(message)s")
```

Comment:
I see you use the same `fmt` as in the `DefaultHandler`. Should we extract this to something like `QiskitNatureLogging.LOG_FORMAT`?
I don't see a good reason for making this configurable but having a fixed constant might be better in case we change this in the future (i.e. only have the string in one place)